symengine/polynomial.cpp

```diff
@@ -20,11 +20,15 @@ UnivariateIntPolynomial::UnivariateIntPolynomial(
     const RCP<const Symbol> &var, const std::vector<integer_class> &v)
     : var_{var}
 {
-    for (unsigned int i = 0; i < v.size(); i++)
-        if (v[i] != 0)
-            dict_add_term(dict_, v[i], i);
-    degree_ = v.size() - 1;
-    SYMENGINE_ASSERT(is_canonical(degree_, dict_))
+    dict_ = {};
+    unsigned int deg = 0;
+    for (unsigned int i = 0; i < v.size(); i++) {
+        if (v[i] != 0) {
+            dict_[i] = v[i];
+            deg = i;
+        }
+    }
+    degree_ = deg;
```

Comment:
Any particular reason? I'll also have to add a check if the dictionary isn't empty too.